src/test/java/org/opensearch/neuralsearch/processor/rerank/MLOpenSearchRerankProcessorIT.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2023 Aryn
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.opensearch.neuralsearch.processor.rerank;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.hc.core5.http.HttpHeaders;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+import org.apache.hc.core5.http.message.BasicHeader;
+import org.junit.After;
+import org.opensearch.client.Request;
+import org.opensearch.client.Response;
+import org.opensearch.common.xcontent.XContentHelper;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.neuralsearch.common.BaseNeuralSearchIT;
+
+import com.google.common.collect.ImmutableList;
+
+import lombok.SneakyThrows;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+public class MLOpenSearchRerankProcessorIT extends BaseNeuralSearchIT {
+
+    final static String PIPELINE_NAME = "rerank-mlos-pipeline";
+    final static String INDEX_NAME = "rerank-test";
+    final static String TEXT_REP_1 = "Jacques loves fish. Fish make Jacques happy";
+    final static String TEXT_REP_2 = "Fish like to eat plankton";
+    final static String INDEX_CONFIG = "{\"mappings\": {\"properties\": {\"text_representation\": {\"type\": \"text\"}}}}";
+
+    @After
+    @SneakyThrows
+    public void tearDown() {
+        super.tearDown();
+        /* this is required to minimize chance of model not being deployed due to open memory CB,
+         * this happens in case we leave model from previous test case. We use new model for every test, and old model
+         * can be undeployed and deleted to free resources after each test case execution.
+         */
+        deleteSearchPipeline(PIPELINE_NAME);
+        findDeployedModels().forEach(this::deleteModel);
```

Comment:
This is little dangerous, as IT can run on remote cluster where different plugins IT runs in parallel it can cause deletion of models which are not uploaded by this plugin. Lets make sure we only delete model that is being uploaded by the test which is running.

Reply:
not sure why I did it this way tbh. will fix

Reply:
found why I did this: https://github.com/opensearch-project/neural-search/blob/91a1202714bbeb58412cb19c069f189c2d6ca3fe/src/test/java/org/opensearch/neuralsearch/processor/TextEmbeddingProcessorIT.java#L38
Should I try to fix there as well?